markovflow/models/spatio_temporal_variational.py

```diff
@@ -434,6 +439,7 @@ def __init__(
             posterior_dist=self.dist_q,
             kernel=self.kernel,
             conditioning_time_points=self.inducing_time,
+            mean_function=self._mean_function.mean_function_time,
```

Comment:
I think this is not the correct way of doing. But I might be wrong
I believe you should add both the spatial and temporal mean function at the very end.
(so remove the temporal mean from the temporal Conditional Process)

like build `f(x,t) + m(x,t)`

right now you distribute the means across the generative model

I think it does not work the way you want.

Let's look at the prior mean `E[f(x,t)]` of the process you create

you build `s(t)` [mean free], then extract such that `E[f(z,t)]=H E[s(t)] + m_t(t)`
then pass this to the spatial conditional `E[f(x,t)] = P E[f(z,t)] + m_s(x)`  (where `P = Kxz Kzz^-1` )
which means you get `E[f(x,t)] = P*H E[s(t)] + P * m_t(t) + m_s(x)`
so the implicit mean function you get is `P * m_t(t) + m_s(x)` instead of `m_t(t) + m_s(x)`

You can try that with a test? 
Like distributing the mean functions vs adding it at the end

Reply:
Thanks, I think you are right, and I'm working on testing this. In the process, I uncovered a bug: #22